crates/nu-command/src/network/url/parse.rs

```diff
@@ -0,0 +1,184 @@
+    let url_string = value.clone().into_string("", engine_state.get_config());
+
+    let result_url = Url::parse(url_string.as_str());
+
+    let head = value.span()?;
+
+    match result_url {
+        Ok(url) => {
+            let cols = vec![
+                String::from("scheme"),
+                String::from("host"),
+                String::from("port"),
+                String::from("path"),
+                String::from("query"),
+                String::from("params"),
+            ];
+            let mut vals: Vec<Value> = vec![];
+
+            vals.push(Value::String {
+                val: String::from(url.scheme()),
+                span: head,
+            });
+            vals.push(Value::String {
+                val: String::from(url.host_str().unwrap_or("")),
+                span: head,
+            });
+
+            match url.port() {
+                None => {
+                    vals.push(Value::String {
+                        val: String::from(""),
+                        span: head,
+                    });
+                }
+                Some(port) => {
+                    vals.push(Value::String {
+                        val: port.to_string(),
+                        span: head,
+                    });
+                }
+            }
+
+            vals.push(Value::String {
+                val: String::from(url.path()),
+                span: head,
+            });
+            vals.push(Value::String {
+                val: String::from(url.query().unwrap_or("")),
+                span: head,
+            });
+
+            let params =
+                serde_urlencoded::from_str::<Vec<(String, String)>>(url.query().unwrap_or(""));
+            match params {
+                Ok(result) => {
+                    let mut param_cols = vec![];
+                    let mut param_vals = vec![];
+                    for (k, v) in result {
+                        param_cols.push(k);
+                        param_vals.push(Value::String { val: v, span: head })
+                    }
+
+                    vals.push(Value::Record {
+                        cols: param_cols,
+                        vals: param_vals,
+                        span: head,
+                    });
+
+                    Ok(PipelineData::Value(
+                        Value::Record {
+                            cols,
+                            vals,
+                            span: head,
+                        },
+                        None,
+                    ))
+                }
+
+                _ => Err(ShellError::UnsupportedInput(
+                    "String not compatible with url-encoding".to_string(),
+                    head,
+                )),
+            }
+        }
+        Err(_e) => {
+            return Err(ShellError::UnsupportedInput(
+                "Incomplete or incorrect URL. Expected a full Url, e.g., https://www.example.com"
```

Comment:
Inconsistent capitalization in error message. (It's nice to have our error messages perfect!)